proposals/public-suffix.md

```diff
@@ -0,0 +1,555 @@
+  // ==> "example.co.uk"
+  //
+  export function getRegistrableDomain(
+    // The domain name whose registrable domain we want to find
+    domain: string,
+    // Options that control the behaviour of the lookup algorithm
+    options?: RegistrableDomainOptions,
+  )
+  // Resolves to the longest registrable domain of the input domain name
+  : Promise<string>;
+
+  //
+  // Gets both registrable domains (i.e. ICANN/unknown, Private) for each domain name
+  // in an array of domain names.
+  //
+  // Example:
+  //
+  // let domains = await browser.publicSuffix.getRegistrableDomains([
+  //   "foo.bar.wixsite.com",
+  //   "www.example.net",
+  //   "a..b",
+  // ]);
+  // ==> [
+  //   { domain: "foo.bar.wixsite.com", base: "wixsite.com", private: "bar.wixsite.com" },
+  //   { domain: "www.example.net",     base: "example.net",                            },
+  //   { domain: "a..b",                error: "Invalid domain name",                   },
+  // ]
+  //
+  export function getRegistrableDomains(
+    // The domain names whose registrable domains we want to find
+    domains: Array<string>,
+    // Options that control the behaviour of the lookup algorithm
+    options?: RegistrableDomainOptions,
+  )
+  // Resolves to the registrable domains of each input domain name
+  : Promise<Array<RegistrableDomains>>;
+
+  //
+  // Gets the PSL dataset version if available
+  //
+  export function getVersion(): string?;
+}
+```
+
+### Behaviours
+
+#### 1. Private Suffixes
+
+By default, the lookup performed by `getRegistrableDomain()` should **include** all
+suffixes in the PSL dataset, i.e. both ICANN and Private (non-ICANN) suffixes.
+
+However, if an `options` object is passed to `getRegistrableDomain()` with key
+`excludePrivateSuffixes` set to `true`, then Private (non-ICANN) suffixes should be
+**excluded** from the lookup algorithm.
+
+##### Example
+
+`domain` = foo.bar.wixsite.com
+
+| Option                                   |   Registrable Domain | PSL Section |
+|------------------------------------------|:--------------------:|:-----------:|
+| excludePrivateSuffixes = false (default) |      bar.wixsite.com | Private     |
+| excludePrivateSuffixes = true            |          wixsite.com | ICANN       |
+
+#### 2. Multiple Suffixes per Domain
+
+The lookup performed by `getRegistrableDomain()` should select the **longest** matching
+suffix (unless specifically excluded using the `excludePrivateSuffixes` option).
+
+##### Example
+
+`domain` = foo.bar.lib.de.us
+
+| Candidate Suffix | PSL Section |
+|-----------------:|:-----------:|
+|            de.us | ICANN       |
+|        lib.de.us | Private     |
+
+The longest is lib.de.us, so `getRegistrableDomain()` resolves to bar.lib.de.us
+
+#### 3. PSL Special Rules
+
+The lookup performed by `getRegistrableDomain()` should adhere to the
+[PSL algorithm](https://github.com/publicsuffix/list/wiki/Format#formal-algorithm).
```

Comment:
I support doing it this way but we will have to talk about it because that is not what browsers are currently doing. 

That situation has been unresolved for a long time and people have been working around it or ignoring it. I suggest we simply defer to what the browsers are already doing here because that is what we want.